File: lib/models/teahouse.dart
class TeahousePostModel {
  final int id;
  final String title;
  final String author;
  final String content;
  final String createdAt;

  TeahousePostModel({
    required this.id,
    required this.title,
    required this.author,
    required this.content,
    required this.createdAt,
  });

  factory TeahousePostModel.fromMap(Map<String, dynamic> m) => TeahousePostModel(
        id: m['id'] as int,
        title: m['title'] as String,
        author: (m['author'] ?? '') as String,
        content: (m['content'] ?? '') as String,
        createdAt: (m['created_at'] ?? '') as String,
      );

  Map<String, dynamic> toMap() => {
        'id': id,
        'title': title,
        'author': author,
        'content': content,
        'created_at': createdAt,
      };
}

class CommentModel {
  final int id;
  final int postId;
  final String author;
  final String content;
  final String createdAt;

  CommentModel({
    required this.id,
    required this.postId,
    required this.author,
    required this.content,
    required this.createdAt,
  });

  factory CommentModel.fromMap(Map<String, dynamic> m) => CommentModel(
        id: m['id'] as int,
        postId: m['post_id'] as int,
        author: (m['author'] ?? '') as String,
        content: (m['content'] ?? '') as String,
        createdAt: (m['created_at'] ?? '') as String,
      );

  Map<String, dynamic> toMap() => {
        'id': id,
        'post_id': postId,
        'author': author,
        'content': content,
        'created_at': createdAt,
      };
}

File: native/hub/src/signals/teahouse.rs
use super::AccountData;
use rinf::{DartSignal, RustSignal, SignalPiece};
use serde::{Deserialize, Serialize};

#[derive(Deserialize, DartSignal)]
pub struct TeahouseListInput {
    pub page: Option<i32>,
    pub page_size: Option<i32>,
    pub account: Option<AccountData>,
}

#[derive(Clone, Serialize, Deserialize, SignalPiece)]
pub struct TeahousePost {
    pub id: String,
    pub title: String,
    pub author: String,
    pub content: String,
    pub created_at: String,
}

#[derive(Serialize, RustSignal)]
pub struct TeahouseListOutput {
    pub ok: bool,
    pub data: Vec<TeahousePost>,
    pub error: Option<String>,
}

#[derive(Deserialize, DartSignal)]
pub struct CreatePostInput {
    pub title: String,
    pub content: String,
    pub account: Option<AccountData>,
}

#[derive(Serialize, RustSignal)]
pub struct CreatePostOutput {
    pub ok: bool,
    pub post: Option<TeahousePost>,
    pub error: Option<String>,
}

#[derive(Clone, Serialize, Deserialize, SignalPiece)]
pub struct Comment {
    pub id: String,
    pub post_id: String,
    pub author: String,
    pub content: String,
    pub created_at: String,
}

#[derive(Deserialize, DartSignal)]
pub struct FetchCommentsInput {
    pub post_id: String,
}

#[derive(Serialize, RustSignal)]
pub struct CommentsListOutput {
    pub ok: bool,
    pub data: Vec<Comment>,
    pub error: Option<String>,
}

#[derive(Deserialize, DartSignal)]
pub struct CreateCommentInput {
    pub post_id: String,
    pub content: String,
    pub account: Option<AccountData>,
}

#[derive(Serialize, RustSignal)]
pub struct CreateCommentOutput {
    pub ok: bool,
    pub comment: Option<Comment>,
    pub error: Option<String>,
}

#[derive(Deserialize, DartSignal)]
pub struct DeletePostInput {
    pub post_id: String,
}

#[derive(Serialize, RustSignal)]
pub struct DeletePostOutput {
    pub ok: bool,
    pub error: Option<String>,
}
